core/local/atom_watcher.js

```diff
@@ -0,0 +1,81 @@
+/* @flow */
+
+const fse = require('fs-extra')
+const Promise = require('bluebird')
+
+const checksumer = require('./checksumer')
+const logger = require('../logger')
+const LinuxSource = require('./layers/linux')
+const Identifier = require('./layers/identifier')
+const ChecksumLayer = require('./layers/checksum')
+const Dispatcher = require('./layers/dispatcher')
+
+/*::
+import type Pouch from '../pouch'
+import type Prep from '../prep'
+import type EventEmitter from 'events'
+import type { Checksumer } from './checksumer'
+*/
+
+const log = logger({
+  component: 'AtomWatcher'
+})
+
+module.exports = class AtomWatcher {
+  /*::
+  syncPath: string
+  events: EventEmitter
+  checksumer: Checksumer
+  ensureDirInterval: *
+  running: Promise<void>
+  _runningResolve: ?Function
+  _runningReject: ?Function
+  source: LinuxSource
+  start: () => Promise<*>
+  stop: (force: ?bool) => Promise<*>
+  ensureDirSync: () => void
```

Comment:
I believe the way the `Watcher` interface is declared is messing up flow (my bad, I didn't noticed it in the previous PR).
Following [flow interface syntax](https://flow.org/en/docs/types/interfaces/#toc-interface-syntax) for methods/props (aka no `:` after method name, `:` instead of `=>` for method return type, end every line with `;`) should allow removing the 3 lines above.